site/assets/js/tac.js

```diff
@@ -44,13 +44,45 @@
     })
   }, { once: true })
 
-  document.addEventListener('googletagmanager_added', function () {
-    var ask = document.querySelector('#googletagmanagerLine .tarteaucitronAsk')
-    var choiceEvent = 'tarteaucitron.userInterface.respond(document.getElementById(\'googletagmanagerAllowed\'),document.getElementById(\'googletagmanagerAllowed\').checked);'
-    var toggle = document.createRange().createContextualFragment('<div class="form-check form-switch my-2 my-lg-0"><input class="form-check-input ms-0" type="checkbox" id="googletagmanagerAllowed" aria-describedby="tacCLgoogletagmanager" onchange="' + choiceEvent + '"><label class="form-check-label visually-hidden" for="googletagmanagerAllowed">Google Tag Manager</label></div>')
-    ask.innerHTML = ''
-    ask.append(toggle)
-  }, { once: true })
+  document.addEventListener(
+    'googletagmanager_added',
+    function () {
+      var ask = document.querySelector(
+        '#googletagmanagerLine .tarteaucitronAsk'
+      )
+      var choiceEvent =
+        'tarteaucitron.userInterface.respond(document.getElementById("googletagmanagerAllowed"),document.getElementById("googletagmanagerAllowed").checked);'
+      var toggle = document
+        .createRange()
+        .createContextualFragment(
+          '<div class="form-check form-switch my-2 my-lg-0"><input class="form-check-input ms-0" type="checkbox" id="googletagmanagerAllowed" aria-describedby="tacCLgoogletagmanager" onchange="' +
+            choiceEvent +
+            '"><label class="form-check-label visually-hidden" for="googletagmanagerAllowed">Google Tag Manager</label></div>'
+        )
+      ask.innerHTML = ''
+      ask.append(toggle)
+
+      var allowConsentButton = document.querySelector('#tarteaucitronAllAllowed')
```

Comment:
The following code (from here until line 85) is triggered by 'googletagmanager_added'. It is OK for now because we handle only one service. Isn't there another way to trigger it? Something more general?